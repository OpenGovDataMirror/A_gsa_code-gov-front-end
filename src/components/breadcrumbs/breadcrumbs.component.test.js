import React from 'react'
import { shallow } from 'enzyme'

import CustomLink from 'components/custom-link'
import Breadcrumbs from 'components/breadcrumbs/breadcrumbs.component'

const props = {
  crumbs: [
    { text: 'text-1', to: '/to-1' },
    { text: 'text-2', to: '/to-2' },
    { text: 'text-3' },
  ],
}

let wrapper
describe('components - Breadcrumbs', () => {
  beforeEach(() => {
    wrapper = shallow(<Breadcrumbs {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should map all the crumbs out', () => {
    expect(wrapper.find('li').length).toBe(props.crumbs.length)
  })

  it('should render a `CustomLink` if a crumb has a `to` property', () => {
    const expected = props.crumbs.filter(x => x.to).length
    expect(wrapper.find(CustomLink).length).toBe(expected)
  })
})
