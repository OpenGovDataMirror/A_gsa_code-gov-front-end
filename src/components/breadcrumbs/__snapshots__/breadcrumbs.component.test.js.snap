// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - Breadcrumbs should render correctly 1`] = `
<div
  className="grid-container"
  id="breadcrumbs"
>
  <nav
    aria-label="Breadcrumbs"
    className="usa-breadcrumb grid-row tablet-lg:margin-top-4 margin-top-2"
  >
    <ol
      className="usa-breadcrumb__list font-body-2xs"
    >
      <li
        className="usa-breadcrumb__list-item"
        key="text-1"
      >
        <Connect(CustomLinkComponent)
          className="usa-breadcrumb__link"
          to="/to-1"
        >
          <span>
            text-1
          </span>
        </Connect(CustomLinkComponent)>
      </li>
      <li
        className="usa-breadcrumb__list-item"
        key="text-2"
      >
        <Connect(CustomLinkComponent)
          className="usa-breadcrumb__link"
          to="/to-2"
        >
          <span>
            text-2
          </span>
        </Connect(CustomLinkComponent)>
      </li>
      <li
        aria-current="page"
        className="usa-breadcrumb__list-item usa-current text-bold"
        key="text-3"
      >
        <span>
          text-3
        </span>
      </li>
    </ol>
  </nav>
</div>
`;
